### PR TITLE
Add link to templates YAML reference documentation

### PR DESCRIPTION
**Pull request by JohnCgp**
Make the reference documentation more discoverable to
people searching for information about Vela templates.

On the heels of #237

### CODE REVIEW

File: content/templates/_index.md [hunk 1]
```diff
@@ -20,6 +20,10 @@ The following YAML tags are not valid inside a template pipeline:
 * `templates:`
 {{% /alert %}}
 
+## Reference documentation
+
+Check out the [YAML reference documentation](https://go-vela.github.io/docs/reference/yaml/templates/) for templates.
```
Comment (wass3r) [inline content/templates/_index.md]:
```suggestion
Check out the [YAML reference documentation](/docs/reference/yaml/templates/) for templates.
```
I would change to make it a relative link. One benefit is you can test it locally without having links jumps from local to external and potentially get different versions.

**Review by wass3r — CHANGES_REQUESTED** [review]

File: content/templates/_index.md (hunk 1 above)
Comment (JohnCgp) [reply content/templates/_index.md]: Good point, done.

**Review by wass3r — APPROVED** [review]

**Comment by wass3r** [comment]
Thanks for the contribution 🙇🏼

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**